src/fp_Rob_fpop.cc

```diff
@@ -37,40 +39,24 @@ void rob_fpop(const double *profil, const int *nbi, const double *lambda_,
 	double max=*maxi;
 
 
-	int minPosition=-10;
-	double minCurrent=100, meanCurrent=0.;
+	double minCurrent=-5;
 
 	
 	CFunctPart *iniFunc = new CFunctPart(min, max);
 	Liste * candidates = new Liste(iniFunc); 
  
-	candidates->add(profil[0], lthreshold, rthreshold, lslope, rslope);
-
-	// Initialize for -10, -1, -10 origine
-	//Liste * Current = candidates;
-	//while(Current != NULL){
-	//	if(Current->cFunc->a2 == 0){
-	//		Current->cFunc->origine=-10;		
-	//	}	
-	//	Current = Current->next;
-	//}
-	
-	//
-	candidates->getMin(&minCurrent, &minPosition, &meanCurrent);
 	cout_n[0] = minCurrent + lambda;
-	origine[0] = minPosition;
-	mean_n[0] = meanCurrent;
-	
-	for ( int t =1; t < nb; t++ ){
-		//cout << "Tour : " << t << "\n";
-		candidates->compare(cout_n[t-1], t);
-		candidates->mergeSimilarElements();	
-		candidates->add(profil[t], lthreshold, rthreshold, lslope, rslope);
-		candidates->mergeSimilarElements();
-		candidates->getMin(&minCurrent, &minPosition, &meanCurrent);
-		cout_n[t] = minCurrent + lambda;
-		origine[t] = minPosition;
-		mean_n[t] = meanCurrent;
+	for ( int t =0; t < nb; t++ ){
+	  if(t>0){
+	    candidates->compare(cout_n[t-1], t);
+	    candidates->mergeSimilarElements();
+	  }
+	  candidates->add(profil[t], lthreshold, rthreshold, lslope, rslope);
+	  if(t>0){
+	    candidates->mergeSimilarElements();
+	  }
```

Comment:
I would avoid the "ifs" in the for loop and keep the initialization before the for loop.
That would be something like:
/* INITIALIZATION FOR t=0 */
candidates->add(profil[0], lthreshold, rthreshold, lslope, rslope);
candidates->getMin(&minCurrent, &minPosition, &meanCurrent);
cout_n[0] = minCurrent + lambda;
origine[0] = minPosition;
mean_n[0] = meanCurrent;

/* MAIN LOOP */
for ( int t =1; t < nb; t++ ){
		candidates->compare(cout_n[t-1], t);
		candidates->mergeSimilarElements();	
		candidates->add(profil[t], lthreshold, rthreshold, lslope, rslope);
		candidates->mergeSimilarElements();
		candidates->getMin(&minCurrent, origine+t, mean_n+t)
		candidates->getNbIntervals(intervals+t)
}

Reply:
my intent was to emphasize the operations which are in common between the first (t=0) and each other iteration -- this results in reduced repetition in the code.
I guess it is a choice of style, so that is fine to revert if you prefer the original.